pkg/virt-handler/virtwrap/isolation/isolation.go

```diff
@@ -51,20 +51,26 @@ type PodIsolationDetector interface {
 }
 
 type socketBasedIsolationDetector struct {
-	socketDir  string
-	controller []string
+	virtShareDir string
+	controller   []string
 }
 
-// NewSocketBasedIsolationDetector takes socketDir and creates a socket based IsolationDetector
+var defaultControllers = []string{"cpu", "cpuacct", "cpuset", "freezer", "memory", "net_cls", "perf_event"}
+
+// NewSocketBasedIsolationDetector takes virtShareDir and creates a socket based IsolationDetector
 // It returns a PodIsolationDetector which detects pid, cgroups and namespaces of the socket owner.
-func NewSocketBasedIsolationDetector(socketDir string) PodIsolationDetector {
-	return &socketBasedIsolationDetector{socketDir: socketDir, controller: []string{"cpu", "cpuacct", "cpuset", "freezer", "memory", "net_cls", "perf_event"}}
+func NewSocketBasedIsolationDetector(virtShareDir string) PodIsolationDetector {
+
+	iso := &socketBasedIsolationDetector{virtShareDir: virtShareDir}
+	iso.controller = append(iso.controller, defaultControllers...)
+	return iso
 }
 
 func SocketFromNamespaceName(baseDir string, namespace string, name string) string {
-	return filepath.Clean(baseDir) + "/" + namespace + "/" + name + "/sock"
+	return filepath.Clean(baseDir) + "/sockets/" + namespace + "_" + name + "_sock"
```

Comment:
Good candidate for path.Join() instead of all those + "/"...

Reply:
yup